java-util/src/main/java/io/druid/java/util/common/granularity/GranularityType.java

```diff
@@ -0,0 +1,262 @@
+ */
+public enum GranularityType
+{
+  SECOND(
+      "'dt'=yyyy-MM-dd-HH-mm-ss",
+      "'y'=yyyy/'m'=MM/'d'=dd/'h'=HH/'m'=mm/'s'=ss",
+      "'y'=yyyy/'m'=MM/'d'=dd/'H'=HH/'M'=mm/'S'=ss",
+      6,
+      "PT1S"
+  ),
+  MINUTE(
+      "'dt'=yyyy-MM-dd-HH-mm",
+      "'y'=yyyy/'m'=MM/'d'=dd/'h'=HH/'m'=mm",
+      "'y'=yyyy/'m'=MM/'d'=dd/'H'=HH/'M'=mm",
+      5,
+      "PT1M"
+  ),
+  FIVE_MINUTE(MINUTE, "PT5M"),
+  TEN_MINUTE(MINUTE, "PT10M"),
+  FIFTEEN_MINUTE(MINUTE, "PT15M"),
+  THIRTY_MINUTE(MINUTE, "PT30M"),
+  HOUR(
+      "'dt'=yyyy-MM-dd-HH",
+      "'y'=yyyy/'m'=MM/'d'=dd/'h'=HH",
+      "'y'=yyyy/'m'=MM/'d'=dd/'H'=HH",
+      4,
+      "PT1H"
+  ),
+  SIX_HOUR(HOUR, "PT6H"),
+  DAY(
+      "'dt'=yyyy-MM-dd",
+      "'y'=yyyy/'m'=MM/'d'=dd",
+      "'y'=yyyy/'m'=MM/'d'=dd",
+      3,
+      "P1D"
+  ),
+  WEEK(DAY, "P1W"),
+  MONTH(
+      "'dt'=yyyy-MM",
+      "'y'=yyyy/'m'=MM",
+      "'y'=yyyy/'m'=MM",
+      2,
+      "P1M"
+  ),
+  QUARTER(MONTH, "P3M"),
+  YEAR(
+      "'dt'=yyyy",
+      "'y'=yyyy",
+      "'y'=yyyy",
+      1,
+      "P1Y"
+  ),
+  ALL(new AllGranularity()),
+  NONE(new NoneGranularity());
+
+  private final String hiveFormat;
+  private final String lowerDefaultFormat;
+  private final String defaultFormat;
+  private final int dateValuePositions;
+  private final Period period;
+  private final Granularity defaultGranularity;
+
+  GranularityType(Granularity specialGranularity)
+  {
+    this.hiveFormat = null;
+    this.lowerDefaultFormat = null;
+    this.defaultFormat = null;
+    this.dateValuePositions = 0;
+    this.period = null;
+    this.defaultGranularity = specialGranularity;
+  }
+
+  GranularityType(
+      final String hiveFormat,
+      final String lowerDefaultFormat,
+      final String defaultFormat,
+      final int dateValuePositions,
+      final String period
+  )
+  {
+    this.hiveFormat = hiveFormat;
+    this.lowerDefaultFormat = lowerDefaultFormat;
+    this.defaultFormat = defaultFormat;
+    this.dateValuePositions = dateValuePositions;
+    this.period = new Period(period);
+    this.defaultGranularity = new PeriodGranularity(this.period, null, null);
+  }
+
+  GranularityType(GranularityType granularityType, String period)
+  {
+    this(
+        granularityType.getHiveFormat(),
+        granularityType.getLowerDefaultFormat(),
+        granularityType.getDefaultFormat(),
+        granularityType.dateValuePositions,
+        period
+    );
+  }
+
+  Granularity create(DateTime origin, DateTimeZone tz)
+  {
+    if (period != null && (origin != null || tz != null)) {
+      return new PeriodGranularity(period, origin, tz);
+    } else {
+      // If All or None granularity, or if origin and tz are both null, return the cached granularity
+      return defaultGranularity;
+    }
+  }
+
+  public Granularity getDefaultGranularity()
+  {
+    return defaultGranularity;
+  }
+
+  public static DateTime getDateTime(GranularityType granularityType, Integer[] vals)
```

Comment:
This method need not to be static, it should be an instance method of `GranularityType`